---
title       : Gviz - Genomic Region Visualization in R
subtitle    : MonBUG Meeting January 2017
author      : Jean Monlong
job         : PhD Candidate McGill University
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      #
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
output: ioslides_presentation
knit        : slidify::knit2slides
---

```{r, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE, fig.height=3)
```


## R, Bioconductor, Gviz

+ [R](https://www.r-project.org/) : free software environment for statistical computing and graphics.
+ [Bioconductor](http://bioconductor.org/): R package for bioinformatics.
+ [Gviz on Bioconductor](https://bioconductor.org/packages/release/bioc/html/Gviz.html).

## Example

```{r, echo=FALSE, fig.height=5}
load("demo.RData")
from = 36400000
to =   37000000
plotTracks(list(ideoTrack, gatrack, genetrack, atrack, qtrack), from=from, to=to)
```

## Important functions

+ `GenomeAxisTrack`: An axis
+ `IdeogramTrack`: Chromosome ideogram
+ `DataTrack`: Different types of graphs
+ `AnnotationTrack`: Ranges annotation
+ `GeneRegionTrack`: Same but for genes

## Simple example

```{r, echo=FALSE}
library(GenomicRanges)
gr = GRanges("chr21", IRanges(runif(3,36400000,37000000), width=c(1e4,10e4,14e4)), name=c("Mon","BUG","Jan"))
```

```{r}
## Let's say we have 'gr' and 'qtls' GRanges
library(Gviz)
atrack = AnnotationTrack(gr, name="Track")
gatrack = GenomeAxisTrack()
qtrack = DataTrack(qtls, data="logPv", name="QTLs")
plotTracks(list(gatrack, atrack, qtrack))
```

## Data Tracks

```{r, echo=FALSE, out.height=500}
knitr::include_graphics("Gviz-Datatracks.png")
```

## Data track

```{r, echo=FALSE}
qtls = GRanges("chr21", IRanges(sort(unique(runif(1e3,36400000,37000000))), width=1))
qtls$logPv = -log10(runif(length(qtls)))
qtls$logPv[sample.int(length(qtls),3)] = runif(3, 1,10)
```

```{r}
qtrack = DataTrack(qtls, data="logPv", name="QTLs", type="l")
plotTracks(list(atrack, qtrack))
```

## Chromosome ideogram and axis

```{r}
ideoTrack <- IdeogramTrack(genome = "hg19", chromosome = "chr21")
gatrack = GenomeAxisTrack()
plotTracks(list(ideoTrack, gatrack,atrack, qtrack))
```

## Gene model

I downloaded Gencode annotation and parsed it to look like this

```{r}
load("gencode.v25.ch21.RData")
gencode[1]
```

## Gene model track

```{r}
from = 36400000
to =   37000000
genetrack <- GeneRegionTrack(gencode, genome = "hg19", name = "Gene", transcriptAnnotation="symbol")
plotTracks(genetrack, from=from, to=to)
```

## Joining the tracks

```{r}
plotTracks(list(ideoTrack, gatrack, genetrack, atrack, qtrack), from=from, to=to)
```

## UCSC tracks

UCSC tracks can be imported directly, but for some reason it's slow.

```{r, eval=FALSE}
ucscGenes <- UcscTrack(genome="hg19", chromosome="chr21", from=min(start(gr)), to=max(end(gr)), track="knownGene", trackType="GeneRegionTrack", rstarts="exonStarts", rends="exonEnds", gene="name", symbol="name", transcript="name", strand="strand", fill="#8282d2", name="UCSC Genes")
plotTracks(list(gatrack, atrack))
```

## CNV

## Label using the `id` argument

## Color using the `feature` argument

## Group using the `group` argument

## Read alignment

From BAM


## Links

