---
title       : Gviz - Genomic Region Visualization in R
subtitle    : MonBUG Meeting January 2017
author      : Jean Monlong
job         : PhD Candidate McGill University
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      #
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
knit        : slidify::knit2slides
---

```{r, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE, fig.height=3)
```

## R and Bioconductor

+ [R](https://www.r-project.org/) : free software environment for statistical computing and graphics.
+ [Bioconductor](http://bioconductor.org/): R package for bioinformatics.

### GenomicRanges

```{r, echo=FALSE}
library(GenomicRanges)
gr = GRanges("chr21", IRanges(runif(3,36400000,37000000), width=c(1,10,1045)), strand=c("+","-","+"), name=c("Mon","BUG","Jan"))
```

```{r}
gr
```

## Gviz package

```{r}
library(Gviz)
atrack = AnnotationTrack(gr, name="My ranges")
gatrack = GenomeAxisTrack()
plotTracks(list(gatrack, atrack))
```

---

Let's look at some examples:

+ QTL and genes
+ CNV and functional regions.
+ Read alignment

## QTL and gene model

+ Data track: -log10(Pvalues) of the QTL
+ Chromosome ideogram
+ Axis
+ Gene model

## Data track

```{r, echo=FALSE}
qtls = GRanges("chr21", IRanges(unique(runif(1e3,36400000,37000000)), width=1))
qtls$logPv = -log10(runif(length(qtls)))
qtls$logPv[sample.int(length(qtls),3)] = runif(3, 1,10)
```

```{r}
qtrack = DataTrack(qtls, name="QTLs")
plotTracks(list(atrack, qtrack))
```

## Chromosome ideogram and axis

```{r}
ideoTrack <- IdeogramTrack(genome = "hg19", chromosome = "chr21")
plotTracks(list(ideoTrack, gatrack,atrack, qtrack))
```

## Gene model

I downloaded Gencode annotation and parsed it to look like this

```{r}
load("gencode.v25.ch21.RData")
head(gencode)
```

## Gene model track

```{r}
from = 36400000
to =   37000000
genetrack <- GeneRegionTrack(gencode, genome = "hg19", name = "Gene", transcriptAnnotation="symbol")
plotTracks(genetrack, from=from, to=to)
```

---

```{r}
gencode.simp = gencode
gencode.simp$transcript = gencode.simp$symbol
genesimptrack <- GeneRegionTrack(gencode.simp, genome = "hg19", name = "Gene", transcriptAnnotation = "symbol")
plotTracks(genesimptrack, from=from, to=to)
```

## Joining the tracks

```{r}
plotTracks(list(ideoTrack, gatrack, genetrack, atrack, qtrack), from=from, to=to)
```

## UCSC tracks

UCSC tracks can be imported directly, but for some reason it's slow.

```{r, eval=FALSE}
ucscGenes <- UcscTrack(genome="hg19", chromosome="chr21", from=min(start(gr)), to=max(end(gr)), track="knownGene", trackType="GeneRegionTrack", rstarts="exonStarts", rends="exonEnds", gene="name", symbol="name", transcript="name", strand="strand", fill="#8282d2", name="UCSC Genes")
plotTracks(list(gatrack, atrack))
```

## CNV

## Label using the `id` argument

## Color using the `feature` argument

## Group using the `group` argument

## Read alignment

From BAM

## Data Tracks

![](Gviz-Datatracks.png)

## Links

+ [Gviz on Bioconductor](https://bioconductor.org/packages/release/bioc/html/Gviz.html).
